canonical/bundle-kubeflow
# ci: refactor scan-images.yaml workflow

This commit refactors the scan images workflow to use the aquasecurity/trivy-action Github action instead of the kubeflow-ci/scripts/images/scan-images.sh. This to make the workflow faster and more configurable based on the action's inputs. This commit also applies a mitigation for the upstream issue aquasecurity/trivy-action#389, as the number of DB download requests would be reduced (each image is scanned independently in its own Github runner).

#### Testing instructions

1. Go to Actions
2. Click on "Scan images"
3. Run the workflow using the workflow dispatch from branch `KF-6285-fix-1080`
4. See results

NOTE: there are a couple images that cannot be scanned, either because of [the image does not exist](https://github.com/canonical/bundle-kubeflow/actions/runs/11178975987/job/31077725628#step:5:26), or because there is an issue [accessing a binary/library](https://github.com/canonical/bundle-kubeflow/actions/runs/11178975987/job/31077740971#step:5:104). These issues must be resolved separately.
